app/models/searchgov_url.rb

```diff
@@ -178,13 +195,28 @@ def url_without_protocol
 
   def parse_document
     Rails.logger.info "[SearchgovUrl] Parsing document for #{url}"
-    if /^application|text\/plain/ === response.content_type.mime_type
+    if application_document?
       ApplicationDocument.new(document: download.open, url: url)
     else
       HtmlDocument.new(document: response.to_s, url: url)
     end
   end
 
+  def skip_save?(doc)
+    if application_document?
+      # If an application document, skip if that document's entity tag is unchanged and if
+      # the document was parsed with our current Tika version
+      (doc&.Etag == response.headers.to_hash['Etag']) && (doc&.tika_version == Tika.tika_version)
```

Comment:
Hmm - apparently the world is moving toward downcased headers: https://blog.yaakov.online/http-2-header-casing/. The `httprb` gem (used for our fetching) returns the camel-cased headers by default, but we can use lowercase, symbol accessors instead of converting them to a hash:
```ruby
response.headers[:etag]
```
Let's use the lowercased version of 'Etag' and any other header keys we need. This will also help with consistency in case we ever swap out `httprb` for a different HTTP client, such as Faraday:
```ruby
> Faraday.get('https://search.gov').headers
=> {"content-type"=>"text/html; charset=utf-8",
 "content-length"=>"7431",
 "connection"=>"close",
 "date"=>"Thu, 15 Sep 2022 13:30:59 GMT",
 "accept-ranges"=>"bytes",
 ...
 ```